server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteServicesImpl.java

```diff
@@ -73,12 +113,33 @@ public CheckinNote update(CheckinNote checkinNote) {
     @Override
     public Set<CheckinNote> findByFields(UUID checkinid, UUID createbyid) {
         Set<CheckinNote> checkinNote = new HashSet<>();
+        String workEmail = securityService!=null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail!=null? currentUserServices.findOrSaveUser(null, workEmail) : null;
+        Boolean isAdmin = securityService!=null ? securityService.hasRole(RoleType.Constants.ADMIN_ROLE) : false;
+
         checkinNoteRepository.findAll().forEach(checkinNote::add);
-        if (checkinid != null) {
-            checkinNote.retainAll(checkinNoteRepository.findByCheckinid(checkinid));
-        } else if (createbyid != null) {
-            checkinNote.retainAll(checkinNoteRepository.findByCreatedbyid(createbyid));
-        }
+
+            if(checkinid!=null) {
+                CheckIn checkinRecord = checkinRepo.findById(checkinid).orElse(null);
+                final UUID pdlId = checkinRecord!=null?checkinRecord.getPdlId():null;
+                final UUID teamMemberId = checkinRecord!=null?checkinRecord.getTeamMemberId():null;
+                validate(!currentUser.getId().equals(pdlId)&&!currentUser.getId().equals(teamMemberId)&&!isAdmin,"User is unauthorized to do this operation");
+                checkinNote.retainAll(checkinNoteRepository.findByCheckinid(checkinid));
+
+            } else if(createbyid!=null) {
+                MemberProfile memberRecord = memberRepo.findById(createbyid).orElse(null);
+                validate(!currentUser.getId().equals(memberRecord.getId())&&!isAdmin,"User is unauthorized to do this operation");
+                checkinNote.retainAll(checkinNoteRepository.findByCreatedbyid(createbyid));
+
+            } else {
+                validate(!isAdmin,"User is unauthorized to do this operation");
+            }
         return checkinNote;
     }
+
+    private void validate(@NotNull boolean isError, @NotNull String message, Object... args) {
+        if(isError) {
+            throw new CheckinNotesBadArgException(String.format(message, args));
+        }
```

Comment:
It appears that we have a validate method in each of these validation stories. Is this something that we can pull out and just use everywhere?

Reply:
each validation method is throwing a different exception so I didn't pull it out.